examples/fit-model-ceres.cpp

```diff
@@ -258,7 +258,7 @@ int main(int argc, char *argv[])
 	Problem camera_costfunction;
 	for (int i = 0; i < image_points.size(); ++i)
 	{
-		CostFunction* cost_function = new AutoDiffCostFunction<fitting::LandmarkCost, 2 /* num residuals */, 4 /* camera rotation (quaternion) */, num_cam_trans_intr_params /* camera translation & fov/frustum_scale */, 10 /* shape-coeffs */, 6 /* bs-coeffs */>(new fitting::LandmarkCost(morphable_model.get_shape_model(), blendshapes, image_points[i], vertex_indices[i], image.cols, image.rows, use_perspective));
+		CostFunction* cost_function = new AutoDiffCostFunction<fitting::LandmarkCost, 2 /* num residuals */, 4 /* camera rotation (quaternion) */, num_cam_trans_intr_params /* camera translation & fov/frustum_scale */, 10 /* shape-coeffs */, 6 /* bs-coeffs */>(new fitting::LandmarkCost(morphable_model.get_shape_model(), blendshapes, image_points[i], vertex_indices[i], 10, image.cols, image.rows, use_perspective));
```

Comment:
I think the problem here is, that the template parameter `10` is for indicating the array size (==dimension) of input parameter of the shape coefficients when evaluating `operator()` of the `fitting::LandmarkCost` cost function that is put as an argument to the `AutoDiffCostFunction` and only the `AutoDiffCostFunction` knows the size only through the template argument and cannot give the size to the `operator()`. Thus, when trying to evaluate the model, we somehow need to know how many parameters are there and the only possibility would be at the constructor of `fitting::LandmarkCost` I think.